scripts/release-tools/downstream/y-stream-setup.sh

```diff
@@ -0,0 +1,119 @@
+# Extract major and minor version numbers
+previous_major=$(cut -d '.' -f 1 <<< "$previous_release")
+previous_minor=$(cut -d '.' -f 2 <<< "$previous_release")
+release_major=$(cut -d '.' -f 1 <<< "$release")
+release_minor=$(cut -d '.' -f 2 <<< "$release")
+
+# Compare versions
+if [ "$previous_major" -gt "$release_major" ] || { [ "$previous_major" -eq "$release_major" ] && [ "$previous_minor" -gt "$release_minor" ] ; }; then
+    echo "Previous release must be less than the current release."
+    exit 1
+fi
+
+get_supported_versions() {
+    #supported_versions=($(curl -fsSL "https://access.redhat.com/product-life-cycles/api/v1/products?name=Red%20Hat%20Advanced%20Cluster%20Security%20for%20Kubernetes" | jq -r '.data[0].versions[] | select(.type == "Full Support") | .name'))
+
+    mapfile -t supported_versions < <(
+      curl -fsSL "https://access.redhat.com/product-life-cycles/api/v1/products?name=Red%20Hat%20Advanced%20Cluster%20Security%20for%20Kubernetes" |
+      jq -r '.data[0].versions[] | select(.type == "Full Support") | .name'
+    )
+
+
+    nversions=${#supported_versions[@]}
+    for ((i = nversions - 1; i >= 0; i = i - 1)); do
+        echo "${supported_versions[$i]}"
+    done
+
+    echo "$release"
+}
+
+update_content_stream_tags() {
+    mapfile -t versions < <(get_supported_versions)
+
+    nversions=${#versions[@]}
+
+    for ((i=0; i<nversions; i++)); do
+        find versions/release-* -name 'product.yml' -exec bash -c 'yq w -i "$1" delivery-repo-content.content_stream_tags[$2] "$3" --style=double' _ {} "$i" "${versions[$i]}" \;
+    done
+}
+
+perform_sed_or_gsed() {
+    local search="$1"
+    local replace="$2"
+    local file="$3"
+
+    if command -v sed > /dev/null; then
+        sed -i "s|$search|$replace|" "$file"
+    elif command -v gsed > /dev/null; then
+        gsed -i "s|$search|$replace|" "$file"
+    else
+        echo "Error: Neither sed nor gsed found. Cannot perform replacement."
+        exit 1
+    fi
+}
+
+export -f perform_sed_or_gsed
+
+git clone git@gitlab.cee.redhat.com:cpaas-products/rhacs.git gitlab-rhacs
+pushd gitlab-rhacs
+
+git checkout -b "setup-${release}"
+
+cp -R "versions/release-${previous_release}" "versions/release-${release}"
+
+pushd "versions/release-${release}"
+
+rm -f advisory_map.yml
+
+# Update release.yml with the correct version. Sinple sed should be safe, but check changes
+sed -i "s|$previous_release|$release|" release.yml
+
+# Update product.yml with the correct version.
+# This is a little more complicated since not all occurances of the old release should be changed
+# Check changes
+perform_sed_or_gsed "rhacs-$previous_release" "rhacs-$release" product.yml
+perform_sed_or_gsed "RHACS-$previous_release" "RHACS-$release" product.yml
+perform_sed_or_gsed "RHACS $previous_release" "RHACS $release" product.yml
+perform_sed_or_gsed "Kubernetes $previous_release" "Kubernetes $release" product.yml
+
+yq w -i product.yml product.release.version "${release}.0" --style=single
```

Comment:
We should check at the beginning of the script if `yq` is available and quit with error if not.

Reply:
Done